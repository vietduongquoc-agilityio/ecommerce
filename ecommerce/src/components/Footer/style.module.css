.footer {
  background-color: var(--background);
  padding: 40px 0;
  margin-top: 100px;
}

.heading {
  font-size: 18px;
  font-weight: bold;
  margin-bottom: 8px;
}

.link {
  color: var(--foreground);
  text-decoration: none;
  margin: 4px 0;
  font-size: 16px;
  font-weight: 500;
}

.link:hover {
  text-decoration: underline;
}

.form {
  display: flex;
  gap: 8px;
}

.input {
  font-weight: 400;
  font-size: 14px;
  border: none;
}

.input:hover {
  text-decoration: underline;
}

.button {
  padding: 8px 16px;
  background-color: var(--foreground);
  color: var(--background);
  border: none;
  cursor: pointer;
}

.button:hover {
  background-color: #444;
}

.copyright {
  font-size: 16px;
  font-weight: 400;
  padding-top: 30px;
  border-top: solid 1px #D9D9D9;
  margin-top: 10px;
}

.address {
  color: #9f9f9f;
  font-size: 16px;
  font-weight: 400;
}

@media (prefers-color-scheme: dark) {
  .footer {
    background-color: #0a0a0a;
  }
  .link,
  .heading,
  .copyright {
    color: #9f9f9f;
  }
  .input {
    border: 1px solid #ededed;
  }
  .button {
    background-color: #ededed;
    color: #0a0a0a;
  }
}

