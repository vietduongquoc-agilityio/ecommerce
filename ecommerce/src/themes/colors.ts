const colors = {
  white: "#FFFFFF",

  primary: {
    yellow: "#B88E2F",
    light: "#FFF3E3",
    floral: "#FCF8F3"
  },

  secondary: {
    dark: "#333333",
    black: "#3A3A3A",
    granite: "#666666",
    light: "#D9D9D9",
    linen: "#FAF3EA",
    splight: "#9F9F9F"
  },

  success: {
    50: "#e7f8f0",
    400: "#41c588",
  },

  error: {
    50: "#feeceb",
    400: "#f36960",
  },

  warning: {
    50: "#fef4e6",
    400: "#f9A63A",
  },
};

export default colors;

