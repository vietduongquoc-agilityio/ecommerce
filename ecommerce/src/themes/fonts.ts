const fonts = {
  family: "Poppins, sans-serif",
  size: {
    xs: "16px",
    sm: "18px",
    md: "20px",
    lg: "24px",
    xl: "34px",
    "2xl": "52px",
  },
  weight: {
    regular: 400,
    medium: 500,
    semiBold: 600,
    bold: 700,
  },
  borderRadius: {
    sm: "10px",
    md: "15px",
    lg: "50px",
  },
};

export default fonts;
