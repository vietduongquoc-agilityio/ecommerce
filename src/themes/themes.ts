import { colors, fonts } from ".";

export const theme = {
  ...colors,
  ...fonts,
  styles: {
    global: {
      "html, body": {
        fontFamily: "body",
        bg: "white",
      },
    },
  },
};

export default theme;
