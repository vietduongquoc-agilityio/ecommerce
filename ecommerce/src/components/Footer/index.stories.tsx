import React from "react";
import Footer from "@/components/Footer";
import { Theme } from "@radix-ui/themes";
import type { Meta, StoryObj } from "@storybook/react";
import "@/components/Footer/style.module.css";

const meta: Meta<typeof Footer> = {
  title: "Components/Footer",
  component: Footer,
  parameters: {
    layout: "fullscreen",
  },
};

export default meta;

type Story = StoryObj<typeof Footer>;

export const Default: Story = {
  render: () => (
    <Theme>
      <Footer />
    </Theme>
  ),
};
