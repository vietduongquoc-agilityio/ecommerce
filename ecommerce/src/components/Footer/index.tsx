"use client";

import { Flex, Text } from "@radix-ui/themes";
import { colors, fonts } from "@/themes";

export default function Footer() {
  return (
    <footer className="footer">
      <Flex
        width="100%"
        maxWidth="1134px"
        style={{ borderBottom: "solid 1px #D9D9D9", paddingBottom: "10px" }}
      >
        <Flex direction="column" mr="136px">
          <Text
            className="heading"
            style={{ fontSize: fonts.size.lg, fontWeight: fonts.weight.bold }}
          >
            Funiro.
          </Text>
          <Text
            className="address"
            style={{
              width: "285px",
              fontSize: fonts.size.xs,
              fontWeight: fonts.weight.regular,
              color: colors.secondary.splight,
              marginTop: "50px",
            }}
          >
            400 University Drive Suite 200 Coral Gables,
            <div>FL 33134 USA </div>
          </Text>
        </Flex>
        <Flex
          direction="column"
          mr="144px"
          gap="46px"
          style={{
            fontSize: fonts.size.xs,
            fontWeight: fonts.weight.medium,
          }}
        >
          <Text className="address" style={{ color: colors.secondary.splight }}>
            Links
          </Text>
          <p className="link">Home</p>
          <p className="link">Shop</p>
          <p className="link">About</p>
          <p className="link">Contact</p>
        </Flex>
        <Flex
          direction="column"
          mr="72px"
          gap="46px"
          style={{
            fontSize: fonts.size.xs,
            fontWeight: fonts.weight.medium,
          }}
        >
          <Text
            className="address"
            style={{ color: colors.secondary.splight, width: "140px" }}
          >
            Help
          </Text>
          <p className="link">Payment Options</p>
          <p className="link">Returns</p>
          <p className="link">Privacy Policies</p>
        </Flex>
        <Flex
          style={{
            fontSize: fonts.size.xs,
            fontWeight: fonts.weight.medium,
          }}
          direction="column"
          gap="46px"
          width="100%"
          maxWidth="286px"
        >
          <Text className="address" style={{ color: colors.secondary.splight }}>
            Newsletter
          </Text>
          <form className="form">
            <input
              type="email"
              placeholder="Enter your email address"
              className="input"
              style={{
                border: "none",
              }}
            />
            <button
              className="button"
              style={{
                padding: "8px 16px",
                background: "none",
                border: "none",
                cursor: "not-allowed",
              }}
            >
              Subscribe
            </button>
          </form>
        </Flex>
      </Flex>
      <div
        className="copyright"
        style={{ paddingTop: "20px", paddingBottom: "40px" }}
      >
        2023 Funiro. All rights reserved.
      </div>
    </footer>
  );
}
