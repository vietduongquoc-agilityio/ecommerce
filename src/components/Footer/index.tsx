import { Text } from "@radix-ui/themes";
import { colors, fonts } from "@/themes";

// Components
import { Button } from "..";

const Footer = () => {
  return (
    <footer
      className="footer"
      style={{
        width: "100%",
        borderTop: "solid 1px #D9D9D9",
      }}
    >
      <div
        style={{
          width: "100%",
          maxWidth: "1133px",
          display: "flex",
          margin: "0 auto",
          padding: "48px 0",
        }}
      >
        <div
          style={{
            display: "flex",
            flexDirection: "column",
            marginRight: "136px",
          }}
        >
          <Text
            className="heading"
            style={{ fontSize: fonts.size.lg, fontWeight: fonts.weight.bold }}
          >
            Funiro.
          </Text>
          <Text
            className="address"
            style={{
              width: "285px",
              fontSize: fonts.size.xs,
              fontWeight: fonts.weight.regular,
              color: colors.secondary.splight,
              marginTop: "50px",
            }}
          >
            400 University Drive Suite 200 Coral Gables,
            <div>FL 33134 USA </div>
          </Text>
        </div>
        <div
          style={{
            fontSize: fonts.size.xs,
            fontWeight: fonts.weight.medium,
            display: "flex",
            flexDirection: "column",
            marginRight: "144px",
            gap: "46px",
          }}
        >
          <Text className="address" style={{ color: colors.secondary.splight }}>
            Links
          </Text>
          <p className="link">Home</p>
          <p className="link">Shop</p>
          <p className="link">About</p>
          <p className="link">Contact</p>
        </div>
        <div
          style={{
            fontSize: fonts.size.xs,
            fontWeight: fonts.weight.medium,
            display: "flex",
            flexDirection: "column",
            marginRight: "72px",
            gap: "46px",
          }}
        >
          <Text
            className="address"
            style={{ color: colors.secondary.splight, width: "140px" }}
          >
            Help
          </Text>
          <p className="link">Payment Options</p>
          <p className="link">Returns</p>
          <p className="link">Privacy Policies</p>
        </div>
        <div
          style={{
            fontSize: fonts.size.xs,
            fontWeight: fonts.weight.medium,
            display: "flex",
            flexDirection: "column",
            gap: "46px",
          }}
        >
          <Text className="address" style={{ color: colors.secondary.splight }}>
            Newsletter
          </Text>
          <form className="form">
            <input
              type="email"
              placeholder="Enter your email address"
              className="input"
              style={{
                border: "none",
              }}
            />
            <Button variant="ghost">Subscribe</Button>
          </form>
        </div>
      </div>
      <div
        className="copyright"
        style={{
          paddingTop: "20px",
          paddingBottom: "40px",
          borderTop: "solid 1px #D9D9D9",
          maxWidth: "1240px",
          width: "100%",
          margin: "0 auto",
        }}
      >
        2023 Funiro. All rights reserved.
      </div>
    </footer>
  );
};

export default Footer;
