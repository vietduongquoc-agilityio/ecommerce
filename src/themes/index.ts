import colors from "./colors";
import fonts from "./fonts";

export { fonts, colors };
