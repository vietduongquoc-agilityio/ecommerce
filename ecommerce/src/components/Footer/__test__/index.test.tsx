import React from "react";
import { render } from "@testing-library/react";
import Footer from "@/components/Footer";
import { Theme } from "@radix-ui/themes";
import "@testing-library/jest-dom";

describe("Footer component", () => {
  it("renders correctly and matches snapshot", () => {
    const { container } = render(
      <Theme>
        <Footer />
      </Theme>
    );
    expect(container).toMatchSnapshot();
  });
});
