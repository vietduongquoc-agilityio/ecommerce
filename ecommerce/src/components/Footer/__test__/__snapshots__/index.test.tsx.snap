// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footer"
    style="width: 100%; max-width: 1440px; border-top: 1px solid #D9D9D9;"
  >
    <div
      style="padding: 48px 206px 48px 100px; display: flex;"
    >
      <div
        style="display: flex; flex-direction: column; margin-right: 136px;"
      >
        <span
          class="rt-Text heading"
          style="font-size: 24px; font-weight: 700;"
        >
          Funiro.
        </span>
        <span
          class="rt-Text address"
          style="width: 285px; font-size: 16px; font-weight: 400; color: rgb(159, 159, 159); margin-top: 50px;"
        >
          400 University Drive Suite 200 Coral Gables,
          <div>
            FL 33134 USA 
          </div>
        </span>
      </div>
      <div
        style="font-size: 16px; font-weight: 500; display: flex; flex-direction: column; margin-right: 144px; gap: 46px;"
      >
        <span
          class="rt-Text address"
          style="color: rgb(159, 159, 159);"
        >
          Links
        </span>
        <p
          class="link"
        >
          Home
        </p>
        <p
          class="link"
        >
          Shop
        </p>
        <p
          class="link"
        >
          About
        </p>
        <p
          class="link"
        >
          Contact
        </p>
      </div>
      <div
        style="font-size: 16px; font-weight: 500; display: flex; flex-direction: column; margin-right: 72px; gap: 46px;"
      >
        <span
          class="rt-Text address"
          style="color: rgb(159, 159, 159); width: 140px;"
        >
          Help
        </span>
        <p
          class="link"
        >
          Payment Options
        </p>
        <p
          class="link"
        >
          Returns
        </p>
        <p
          class="link"
        >
          Privacy Policies
        </p>
      </div>
      <div
        style="font-size: 16px; font-weight: 500; display: flex; flex-direction: column; gap: 46px; max-width: 286px; width: 100%;"
      >
        <span
          class="rt-Text address"
          style="color: rgb(159, 159, 159);"
        >
          Newsletter
        </span>
        <form
          class="form"
        >
          <input
            class="input"
            placeholder="Enter your email address"
            type="email"
          />
          <button
            style="padding: 1px; font-size: 16px; font-weight: 500; text-align: center; width: auto; cursor: not-allowed; transition: all 0.3s ease; background-color: transparent; color: rgb(58, 58, 58);"
          >
            Subscribe
          </button>
        </form>
      </div>
    </div>
    <div
      class="copyright"
      style="padding-top: 20px; padding-bottom: 40px; border-top: 1px solid #D9D9D9; max-width: 1240px; width: 100%; margin: 0px auto;"
    >
      2023 Funiro. All rights reserved.
    </div>
  </footer>
</DocumentFragment>
`;
